NativeApp/src/components/Booking/container.js

```diff
@@ -115,17 +115,16 @@ export default Container => class extends Component {
       });
   }
 
-  updateHoliday = (cancel) => {
-    const { booking, user } = this.state;
+  updateHoliday = () => {
+    const { booking } = this.state;
+    const { endDate, halfDay, startDate, holId } = booking;
```

Comment:
You can destructure this on one line

`const { booking: { endDate, halfDay, startDate, holId } } = this.state;`